### PR TITLE
feat(feedback): Addressing issues in OPC Feedback Component

### LINKED ISSUE #801: Enhancements to the OPC Feedback Component
This issue includes changes to the OPC Feedback Component:

## Configurable Buttons

- **Objective:** Allow customization of "View Feedback" and "Share Feedback" buttons.
- **Desired Outcome:** Users can adjust button behavior and appearance based on specific needs.

## Summary Limit Configuration

- **Objective:** Enable users to set character limits for feedback summaries.
- **Desired Outcome:** Provide adaptability to diverse project needs.

## Reset State After Feedback

- **Objective:** Automatically reset the OPC Feedback Component after feedback submission, eg the text in the summary text box should be reset after submitting a feedback.
- **Desired Outcome:** Ensure a user-friendly and consistent interface.

### PR DESCRIPTION
**Pull request by yashoswalyo**
- fixes #801

# Closes/Fixes/Resolves

### 
Fixes #801 

# Explain the feature/fix

1. Made `spa` and `docs` 
2. Introduced a new attribute `summaryLimit` so that feedback/bug summary is limited.
3. Fixed the <textarea/> state was not getting reset.
4. User can create `onSubmit` function on `opc-feedback` element, which can be used to validate api calls

## Does this PR introduce a breaking change

No

## Screenshots

<details>
<summary>View Screenshots</summary>

image

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] My changes generate no new warnings
- [x] I have performed a self-review of my own code
- [ ] I have documented my code, particularly in areas like todo, complex logic, quick fix, temporary patch, etc.
#### If it is a new component
- [ ] Does your component follow One Platform style guidelines?
- [ ] Does your component web accessibility standards? [Helper Doc](https://www.w3.org/TR/wai-aria-1.1/)
- [ ] Does your component support desktop screen sizes (350px, 720px, 1150px ,1920px)
#### Browsers you have tested in
- [ ] Latest two versions of Mozilla Firefox and Google Chrome supported by Red Hat Enterprise Linux distribution
- [x] Google Chrome [Latest 2 versions]
- [x] Mozilla Firefox [Latest 2 versions]
- [ ] Microsoft Edge [Latest 2 versions]
- [ ] Apple Safari [Latest 2 versions]
- [ ] Microsoft Internet Explorer 11

### CODE REVIEW

**Review by deshmukhmayur — APPROVED** [review]
LGTM.

**Comment by hybridx** [comment]
@yashoswalyo Please bump the version as well, since these have minor updates and no breaking changes please bump the minor version for this.

File: packages/opc-feedback/src/opc-feedback.ts [hunk 1]
```diff
@@ -27,11 +27,14 @@ import {
 import dialogPolyfill from 'dialog-polyfill';
 @customElement('opc-feedback')
 export class OpcFeedback extends LitElement {
-  @property({ type: String, attribute: 'spa' }) spa = '/feedback';
-  @property({ type: String, attribute: 'docs' }) docs = '/get-started';
+  @property({ type: String, attribute: 'spa' }) spa = undefined;
+  @property({ type: String, attribute: 'docs' }) docs = undefined;
```
Comment (hybridx) [inline packages/opc-feedback/src/opc-feedback.ts:31]: @yashoswalyo why were the defaults removed?

**Review by hybridx — CHANGES_REQUESTED** [review]

File: packages/opc-feedback/src/opc-feedback.ts (hunk 1 above)
Comment (yashoswalyo) [reply packages/opc-feedback/src/opc-feedback.ts:31]: To hide it from UI if someone has no SPA for feedback/docs

**Comment by yashoswalyo** [comment]
> @yashoswalyo Please bump the version as well, since these have minor updates and no breaking changes please bump the minor version for this.

Already bumped the version

File: packages/opc-feedback/src/opc-feedback.ts (hunk 1 above)
Comment (hybridx) [reply packages/opc-feedback/src/opc-feedback.ts:31]: Please check https://lit.dev/docs/templates/directives/#ifdefined

**Comment by hybridx** [comment]
> > @yashoswalyo Please bump the version as well, since these have minor updates and no breaking changes please bump the minor version for this.
> 
> I have already bumped the version

I see that you've bumped up the patch, and this seems more like a patch can you update the minor version for this? https://semver.org/

**Comment by yashoswalyo** [comment]
> I see that you've bumped up the patch, and this seems more like a patch can you update the minor version for this? https://semver.org/

Okay so I've bumped it to 0.1.0-alpha

cc @deshmukhmayur

**Review by deshmukhmayur — APPROVED** [review]

File: packages/opc-feedback/src/opc-feedback.ts (hunk 1 above)
Comment (yashoswalyo) [reply packages/opc-feedback/src/opc-feedback.ts:31]: I tried with `ifDefined()`, it renders the element and removes href attribute when `undefined`. 
Which is not the expected outcome.

**Review by riginoommen — APPROVED** [review]

**Comment by riginoommen** [comment]
Can we see why test is failing

**Comment by riginoommen** [comment]
i think its required for publish

**Review by hybridx — APPROVED** [review]

**Comment by hybridx** [comment]
@yashoswalyo There's a different reason for the CI to fail for the mono-repo can you please add s/c for local build and test results. We can go forward with the merge and I can deploy to npm

**Review by hybridx — APPROVED** [review]
Good update LGTM

**Pull request merged**